File: src/ipc/recovery.ts
import { invoke } from "@/ipc/invoke";

/**
 * A document with an unsaved-changes recovery copy on disk. Mirrors
 * `pdf::autosave::RecoveryEntry` on the Rust side.
 *
 * `autosavePath` is the recovery copy under `app_data_dir`; `originalPath`
 * is where the user's file lives. `savedAt` is unix seconds.
 */
export interface RecoveryEntry {
  documentId: string;
  originalPath: string;
  autosavePath: string;
  savedAt: number;
}

/** SPEC: P2.A2 — recoverable documents found at startup. */
export async function recoveryList(): Promise<RecoveryEntry[]> {
  return invoke<RecoveryEntry[]>("recovery_list");
}

/** SPEC: P2.A2 — drop a recovery copy once recovered or declined. */
export async function recoveryDiscard(id: string): Promise<void> {
  return invoke<void>("recovery_discard", { id });
}

File: src-tauri/src/commands/recovery.rs
use tauri::AppHandle;

use crate::error::CommandError;
use crate::pdf::autosave::{self, RecoveryEntry};

/// SPEC: P2.A2 — list documents that have an unsaved-changes recovery
/// copy on disk, so the frontend can offer to reopen them at startup.
#[tauri::command]
pub async fn recovery_list(app: AppHandle) -> Result<Vec<RecoveryEntry>, CommandError> {
    let dir = autosave::autosave_dir(&app)?;
    autosave::scan_autosaves(&dir)
}

/// SPEC: P2.A2 — drop a document's recovery copy after the user has
/// recovered or declined it.
#[tauri::command]
pub async fn recovery_discard(app: AppHandle, id: String) -> Result<(), CommandError> {
    let dir = autosave::autosave_dir(&app)?;
    autosave::discard_autosave(&dir, &id)
}
